test/e2e/apps/job.go

```diff
@@ -823,9 +843,9 @@ done`}
 		job := e2ejob.NewTestJob("fail", "backofflimit", v1.RestartPolicyNever, 1, 1, nil, int32(backoff))
 		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
 		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
-		ginkgo.By("Ensuring job exceed backofflimit")
 
-		err = waitForJobFailure(ctx, f.ClientSet, f.Namespace.Name, job.Name, e2ejob.JobTimeout, "BackoffLimitExceeded")
+		ginkgo.By("Ensuring job exceed backofflimit")
```

Comment:
Can we check the other condition and other status fields here too?

Reply:
Sure, I was thinking about this test too, added in the commit https://github.com/kubernetes/kubernetes/pull/125510/commits/0acffd6f2cfb668b031c0ca49c7ada892da17115